proxy/http/unit_tests/test_http_hook_event_consistency.cc

```diff
@@ -0,0 +1,57 @@
+/** @file
```

Comment:
I think this would be better done with `static_assert` rather than runtime checks.

Reply:
> Are the two commits separable? Or should they be squashed before merging?
> 
> Looking more closely at the individual commits they do look independent, and should not be squashed.

I agree: the first commit is valuable on its own and I'd vote to keep it independent. It fixes a bug in one of our example plugins, for instance. The second commit is dependent upon that first commit, which is why I included it here.

Reply:
Good point. I addressed this here:
https://github.com/apache/trafficserver/pull/6243